src/sequences/Formula.ts

```diff
@@ -2,15 +2,40 @@ import {Cached} from './Cached'
 import {SequenceExportModule} from './SequenceInterface'
 
 import {math} from '@/shared/math'
-import type {GenericParamDescription, ParamValues} from '@/shared/Paramable'
+import type {GenericParamDescription} from '@/shared/Paramable'
 import {ParamType} from '@/shared/ParamType'
+import {ValidationStatus} from '@/shared/ValidationStatus'
 
 const paramDesc = {
     formula: {
         default: 'n',
         type: ParamType.STRING,
         displayName: 'Formula',
         required: true,
+        validate(f: string, status: ValidationStatus) {
+            let parsetree = undefined
+            try {
+                parsetree = math.parse(f)
+            } catch (err: unknown) {
+                status.addError(
+                    `Could not parse formula: ${f}`,
+                    (err as Error).message
+                )
```

Comment:
I'm thinking we should also catch an empty box with "Could not parse formula".  Right now backspacing to empty the box and start afresh (something that will happen plenty) causes an interrupting popup with an unhelpful message (pic attached).  So can we watch out for an empty box before math.js (or whoever is throwing this message) gets involved?
![Screenshot from 2024-10-26 16-35-44](https://github.com/user-attachments/assets/e16e1b47-8e6c-40c7-a165-479ea2a45cb4)

Reply:
Also weird side-effect of this popup is that the numberscope gets in a sensitive state and pops this up more often, even after you've cleared it.  For example, changing n to n+log(2) is fine with a fresh scope, but after this has happened, that same change will cause a pop-up on the final parenthesis.